animation_nodes/nodes/falloff/mesh_falloff.py

```diff
@@ -0,0 +1,64 @@
+import bpy
+from bpy.props import *
+from mathutils.bvhtree import BVHTree
+from ... events import propertyChanged
+from ... base_types import AnimationNode
+from . mesh_falloff_utils import calculateMeshSurfaceFalloff, calculateMeshVolumeFalloff
+
+modeItems = [
+    ("SURFACE", "Surface", "Falloff w.r.t mesh surface", "NONE", 0),
+    ("VOLUME", "Volume", "Falloff w.r.t mesh volume", "NONE", 1)
+]
+
+class MeshFalloffNode(bpy.types.Node, AnimationNode):
+    bl_idname = "an_MeshFalloffNode"
+    bl_label = "Mesh Falloff"
+    errorHandlingType = "EXCEPTION"
+
+    mode: EnumProperty(name = "Mode", default = "SURFACE",
+        items = modeItems, update = AnimationNode.refresh)
+
+    def create(self):
+        self.newInput("Mesh", "Mesh", "mesh")
```

Comment:
Perhaps we should replace this by a BVH Tree instead.

Reply:
Replaced.